test/benchmarks/Benchmarks.Trace/ElasticsearchBenchmark.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Datadog.Trace;
+using Datadog.Trace.ClrProfiler.Emit;
+using Datadog.Trace.ClrProfiler.Integrations;
+using Datadog.Trace.Configuration;
+using Elasticsearch.Net;
+
+namespace Benchmarks.Trace
+{
+    [MemoryDiagnoser]
+    public class ElasticsearchBenchmark
+    {
+        private static readonly int MdToken;
+        private static readonly IntPtr GuidPtr;
+        private static readonly object Pipeline = new RequestPipeline();
+        private static readonly RequestData Data = new RequestData
+        {
+            Method = HttpMethod.POST,
+            Uri = new Uri("http://localhost/"),
+            PathAndQuery = "PathAndQuery"
+        };
+
+        static ElasticsearchBenchmark()
+        {
+            var settings = new TracerSettings
+            {
+                StartupDiagnosticLogEnabled = false
+            };
+
+            Tracer.Instance = new Tracer(settings, new DummyAgentWriter(), null, null, null);
+
+            var methodInfo = typeof(RequestPipeline).GetMethod("CallElasticsearchAsync", System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Instance);
+
+            MdToken = methodInfo.MetadataToken;
+            var guid = typeof(RequestPipeline).Module.ModuleVersionId;
+
+            GuidPtr = Marshal.AllocHGlobal(Marshal.SizeOf(guid));
+
+            Marshal.StructureToPtr(guid, GuidPtr, false);
+
+            new ElasticsearchBenchmark().CallElasticsearchAsync();
```

Comment:
Do you want to wait on the result before returning? Also, should the synchronous `CallElasticSearch` be warmed up as well?

Reply:
You're right 👍